src/main/java/handlers/GetWithBodyHandler.java

```diff
@@ -1,14 +1,24 @@
 package main.java.handlers;
 
+import main.java.server.HTTPMessageComponents.HTTPMethods;
 import main.java.server.request.Request;
-import main.java.server.Response;
+import main.java.server.response.Response;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
 
 import static main.java.server.HTTPMessageComponents.HTTPSyntax.*;
+import static main.java.server.HTTPMessageComponents.StatusCodes.NOT_ALLOWED;
 import static main.java.server.HTTPMessageComponents.StatusCodes.OK;
 
 public class GetWithBodyHandler implements Handler {
+
     @Override
     public Response generateResponseForRequest(Request request) {
+        if (!methodIsAllowed(request.getRequestMethod())) {
+            return new Response(VERSION + SP + NOT_ALLOWED + CRLF, buildHeaders());
+        }
```

Comment:
This looks like it might need to be duplicated in every handler, or perhaps duplicated in every handler _except_ those that handle every method 😱 

I might think about whether we can do this work in one place.